### ISSUE TITLE
Lazy loaded Modal in ionic 4

### ISSUE THREAD

**Issue by trainingnote** [ionic-team/ionic#14581]
# Feature Request

Would it be possible to have ModalController with lazy loading in ionic v4 ? As for now in v4 the only way to create a modal is to use the class instance.

Thank you very much !

**Comment by cvaliere** [comment]
@brandyscarney hi, sorry to ping you like this, but I really think this deserves an answer. It's clearly an important feature that has been lost in migration

**Comment by bashoogzaad** [comment]
I totally agree, I think it is very useful to lazy load a component in a modal.

**Comment by cvaliere** [comment]
thanks for your answer

it seems to me, though, that it would be possible to implement a routing logic inside the modal
the modal could contain its own `<router-outlet>`
opening a modal would change the route of that router-outlet, and navigate to a made-up route corresponding to the component we want to display

am I wrong in this logic?
